api/job_resources.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type JobResources struct {
```

Comment:
Why `JobResources` and not just `Resources`. The object can list more than just jobs.